### PR TITLE
V0.2 Add github workflow integration

### PR DESCRIPTION
**Pull request by romic7x**
# Description

Add integration test with Github workflow for grass-azure pipeline

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [X] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [X] Linux
- Python version:
  - [ ] 3.6
  - [X] 3.7
- Key information snapshot(s):
  
## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

**Review by romic7x — COMMENTED** [review]
This is the test-1 comment submit

**Review by romic7x — COMMENTED** [review]
This is the test-2 comment submit

**Review by romic7x — COMMENTED** [review]
This is the test-3 comment submit

File: maro/cli/grass/executors/grass_executor.py [hunk 1]
```diff
@@ -218,29 +222,68 @@ def remote_update_image_files_details(self):
...
-            "ssh -o StrictHostKeyChecking=no "
+            f"ssh -o StrictHostKeyChecking=no -p {self.ssh_port} "
             f"{self.admin_username}@{self.cluster_details['master']['public_ip_address']} "
             f"'cd {GlobalPaths.MARO_GRASS_LIB}; python3 -m scripts.update_node_status "
             f"{self.cluster_name} {node_name} {action}'"
         )
         _ = SubProcess.run(command)
 
-    def test_connection(self, node_ip_address: str):
+    def test_ssh_22_connection(self, node_ip_address: str):
```
Comment (kaiqli) [inline maro/cli/grass/executors/grass_executor.py:232]: Why have this test separately?

File: maro/cli/grass/executors/grass_executor.py [hunk 2]
```diff
@@ -218,29 +222,68 @@ def remote_update_image_files_details(self):
...
+                self.test_ssh_22_connection(node_ip_address=node_ip_address)
+                self.remote_set_ssh_port(node_ip_address=node_ip_address)
                 return True
             except CliError:
                 remain_retries -= 1
-                logger.debug(f"Unable to connect to {node_ip_address}, remains {remain_retries} retries.")
-                time.sleep(10)
-                continue
+                logger.debug(
+                    f"Unable to connect to {node_ip_address} with port 22, "
```
Comment (kaiqli) [inline maro/cli/grass/executors/grass_executor.py:284]: in one line

File: maro/cli/grass/utils/copy.py [hunk 3]
```diff
@@ -16,28 +16,36 @@
 logger = CliLogger(name=__name__)
 
 
-def copy_files_to_node(local_path: str, remote_dir: str, admin_username: str, node_ip_address: str) -> None:
+def copy_files_to_node(
+    local_path: str, remote_dir: str,
```
Comment (ArthurJiang) [inline maro/cli/grass/utils/copy.py:20]: one line one parameter

File: tests/requirements.test.txt [hunk 4]
```diff
@@ -17,3 +17,4 @@ torch
 pytest
 coverage
 paramiko==2.7.2
+pytz==2019.3
```
Comment (ArthurJiang) [inline tests/requirements.test.txt:20]: when we use it?

**Review by ArthurJiang — DISMISSED** [review]

File: tests/requirements.test.txt (hunk 4 above)
Comment (romic7x) [reply tests/requirements.test.txt:20]: force replace the newer version, the tests only work when pytz < 2020

File: maro/cli/grass/lib/deployments/external/grass_azure_create.yml [hunk 5]
```diff
@@ -16,3 +16,7 @@ master: # MARO Master config
     port: 6379 # [Optional] str: port of the redis server, default as 6379
   samba: # [Optional]
     password: "" # [Optional] str: password of the samba server, default as random string
+
+connection: # [Optional] connection config
+  ssh: # [Optional]
+    port: 51811  # [Optional] int: port of the ssh server, default as 51811
```
Comment (romic7x) [inline maro/cli/grass/lib/deployments/external/grass_azure_create.yml:22]: set 22 instead

File: maro/cli/utils/params.py [hunk 6]
```diff
@@ -10,6 +10,10 @@ class GlobalParams:
     PARALLELS = 5
     LOG_LEVEL = logging.INFO
 
+    DEFAULT_REDIS_PORT = 6379
+    DEFAULT_FLUENTD_PORT = 24224
+    DEFAULT_SSH_PORT = 51811
```
Comment (romic7x) [inline maro/cli/utils/params.py:15]: set 22 instead

File: maro/cli/grass/executors/grass_executor.py (hunk 1 above)
Comment (romic7x) [reply maro/cli/grass/executors/grass_executor.py:232]: We use 22 as ssh port by default, but users can assign a new port to it.
And when we are using grass/azure mode, we will have to replace port 22 with the new port manually, so we have a test_ssh_22 here and a test_ssh_new_port below

File: maro/cli/grass/utils/copy.py (hunk 3 above)
Comment (romic7x) [reply maro/cli/grass/utils/copy.py:20]: like this? Looks like a waste of space
Screen Shot 2020-12-21 at 21 18 44

**Review by micli — APPROVED** [review]

**Review by micli — APPROVED** [review]

**Review by kaiqli — APPROVED** [review]

**Pull request merged**